tests/unit/blocking_connection_tests.py

```diff
@@ -204,3 +208,33 @@ def test_sleep(self, select_connection_class_mock):
             connection.sleep(0.00001)
 
 
+class _SimpleProxy(object):
+
+    def __init__(self, obj):
+        self._obj = obj
+
+    def __getattribute__(self, item):
+        if item in ('_obj', 'send'):
+            return object.__getattribute__(self, item)
+        return getattr(self._obj, item)
+
+
+class TestChannelCreationOnBrokenConnection(unittest.TestCase):
+
+    def test_that_it_raise_closed_connection(self):
+
+        for error in (errno.EPIPE, errno.ECONNABORTED, errno.ECONNREFUSED):
+            # initiate a connection:
+            conn = blocking_connection.BlockingConnection()
```

Comment:
We can probably handle mock testing at each layer without introducing a new dependency. It doesn't seem to be documented or even very active. I don't  I don't know whether Gavin (@gmr) would agree to this dependency. You might want to check with him before investing the time.